### PR TITLE
write authentication in bucket status

### PR DESCRIPTION
**Pull request by upalinski**

### CODE REVIEW

File: contract/pkg/messages.go [hunk 1]
```diff
@@ -54,3 +55,18 @@ type BucketStatus struct {
 func (s *BucketStatus) RentExpired() bool {
 	return s.RentCoveredUntilMs < uint64(time.Now().UnixMilli())
 }
+
+func (s *BucketStatus) HasWriteAccess(publicKey string) (bool, error) {
+	address, err := types.NewAddressFromHexAccountID(publicKey)
```
Comment (svalasovich) [inline contract/pkg/messages.go:60]: With same seed/secretPhrase system generates different public keys for each sign algorithm. Did you test it or now we will be able to test only sr25519 sign algorithm?

File: contract/pkg/mock/ddc_bucket_contract_mock.go [hunk 2]
```diff
@@ -4,15 +4,18 @@ import (
...
 
+var writerIds = getAccountIDs([]string{"5FJDBC3jJbWX48PyfpRCo7pKsFwSy4Mzj5t39PfXixD5jMgy"})
+
 var buckets = []*pkg.BucketStatus{
-	CreateBucket(1, `{"replication":1}`),
-	CreateBucket(2, `{"replication":2}`),
-	CreateBucket(3, `{"replication":3}`),
+	CreateBucket(1, `{"replication":1}`, writerIds),
+	CreateBucket(2, `{"replication":2}`, writerIds),
+	CreateBucket(3, `{"replication":3}`, writerIds),
```
Comment (svalasovich) [inline contract/pkg/mock/ddc_bucket_contract_mock.go:18]: Will work other tests with mock (JS, Kotlin SDKs)?

File: contract/pkg/messages.go (hunk 1 above)
Comment (upalinski) [reply contract/pkg/messages.go:60]: in unit test below i've used ed25519 keys so it suppose to work with different algorithms. The one requirement here is that writerId should have the same algorithm. At the moment in writerId only owner is added, and polkadot extension uses sr25519, so the same algo should be used for you pub/priv key pair you are going to use in library.

Once there is a grantBucketPermission method that can accept accountId, ed25519 can be supported as well. You just need to pass accountId based on ed25519 public key to the grantBucketPermission method

**Review by svalasovich — APPROVED** [review]

File: contract/pkg/mock/ddc_bucket_contract_mock.go (hunk 2 above)
Comment (upalinski) [reply contract/pkg/mock/ddc_bucket_contract_mock.go]: extended writer ids list to support accounts that are used in JS and Kotlin SDKs

**Pull request merged**